fw/cache.h

```diff
@@ -23,7 +23,16 @@
 
 #include "http.h"
 
+/* Cache modes. */
+typedef enum {
+        TFW_CACHE_UNDEFINED = -1,
+        TFW_CACHE_NONE = 0,
+        TFW_CACHE_SHARD,
+        TFW_CACHE_REPLICA,
+} TfwCacheMode;
```

Comment:
There is no sense to expose the particular caching strategy to the rest of the code. Let's move the enum to cache.c and provide `tfw_cache_enabled()` instead of `tfw_cache_get_mode()`